src/mds/Locker.cc

```diff
@@ -4624,6 +4624,17 @@ bool Locker::simple_sync(SimpleLock *lock, bool *need_issue)
       gather++;
       if (lock->is_cached())
 	invalidate_lock_caches(lock);
+
+      // After a client request is early replied the mdlog won't be flushed
+      // immediately, but before safe replied the request will hold the write
+      // locks. So if the client sends another request to a different MDS
+      // daemon, which then needs to request read lock from current MDS daemon,
+      // then that daemon maybe stuck at most for 5 seconds. Which will lead
+      // the client stuck at most 5 seconds.
+      //
+      // Let's try to flush the mdlog when the write lock is held, which will
+      // release the write locks after mdlog is successfully flushed.
+      mds->mdlog->flush();
```

Comment:
I'm not too aware of the Locker code base.
So, I was wondering if invalidating lock caches would do away with the write lock as well.
And, so, shouldn't the log flush take place before invalidating lock caches.

Reply:
> I'm not too aware of the Locker code base.
> So, I was wondering if invalidating lock caches would do away with the write lock as well.

IMO, the simple_sync() is trying to switch the lock state to LOCK_SYNC, which should only be readable by all replicas, so if it's wrlocked we need to invalidate the lock caches before releasing the wrlocks.

> And, so, shouldn't the log flush take place before invalidating lock caches.

From the code I didn't find this is a necessary and there have many other use cases just like this.